lib/cppapi/InterceptPlugin.cc

```diff
@@ -392,7 +391,12 @@ destroyCont(InterceptPlugin::State *state)
     TSVConnClose(state->net_vc_);
     state->net_vc_ = nullptr;
   }
-  if (!state->timeout_action_) {
+
+  if (state->cont_) {
+    if (state->timeout_action_) {
+      TSActionCancel(state->timeout_action_);
+      state->timeout_action_ = nullptr;
```

Comment:
Cancelling the action in the destroy method makes a lot of sense.  Would the original logic have just leaked the continuation?